### DISCUSSION TITLE
How can I get a 'default' input.conf for uosc?

### DISCUSSION THREAD

**Discussion by Diagonet** [tomasklaen/uosc#823] [Q&A] [answered]
I followed the instructions and installed uosc but it didnt come with an input.conf file. I find it much easier to edit and change the keybinds I want to change than starting from scratch. Also, I can only left click the two boxes on the bottom left for menu and subtitles, when I try clicking the menus they open, it doesnt work, only keyboard commands go through

**Comment by christoph-heinrich** [comment]
> I followed the instructions and installed uosc but it didnt come with an input.conf file. I find it much easier to edit and change the keybinds I want to change than starting from scratch.

Have a look at the [examples in the readme](https://github.com/tomasklaen/uosc#examples).

> Also, I can only left click the two boxes on the bottom left for menu and subtitles, when I try clicking the menus they open, it doesnt work, only keyboard commands go through

We've already had someone with problems clicking on menus and it turned out that they had two different versions of uosc installed at the same time. Please make sure that is not the case here.

**Comment by Diagonet** [comment]
The examples are nice but I was hoping to have a file with ALL the default keybinds for uosc, having to start from scratch and find out each and every cmd is really not nice (but fine if you already had an input.conf from another mpv I suppose). Being able to see everything already being used and just edit the keybinds is way more use friendly

I do have mpv.net installed in another folder, dont think that would affect it? never installed uosc before though

**Reply by hooke007** [reply]
>  have a file with ALL the default keybinds for uosc, having to start from scratch and find out each 

You mean it is difficult to copy these commands? All cmds are recorded here.
https://github.com/tomasklaen/uosc?tab=readme-ov-file#commands

**Reply by Diagonet** [reply]
I'm not sure if I'm being clear enough here but in the README for this project mentions editing the input.conf file, there isn't one for me after installing, I have nothing to edit. It isn't using the bindings I have for mpv.net, so I have no idea what the current keybinds are and that list doesn't cover all mpv commands nor would I want to build the entire input.conf file from nothing

**Reply by hrxn** [reply, marked as answer]
@Diagonet there isn't an `input.conf` file for anyone. You have to create one for yourself.
And `input.conf` works entirely independent of uosc.

The default `input.conf` is this one here;
https://github.com/mpv-player/mpv/blob/master/etc/input.conf

You can basically copy this entire file and use it as your own `input.conf`. Although you'd have only duplicated all the default key bindings if you use it like it is without making any changes.

**Reply by christoph-heinrich** [reply]
I'd assume that mpv.net has a input.conf file, so you could copy that as a starting point.

uosc doesn't bind any keys by default, so if you want an input.conf that covers all default bindings, then create an empty file :laughing:

**Reply by hooke007** [reply]
> cover all mpv commands

mpv's manual lists all commands types. And I do not think who will make the file containing all cmds. It's countless in theory.

**Comment by Diagonet** [comment]
> @Diagonet there isn't an `input.conf` file for anyone. You have to create one for yourself. And `input.conf` works entirely independent of uosc.
> 
> The default `input.conf` is this one here; https://github.com/mpv-player/mpv/blob/master/etc/input.conf
> 
> You can basically copy this entire file and use it as your own `input.conf`. Although you'd have only duplicated all the default key bindings if you use it like it is without making any changes.

Thanks. I know that for most here this might be obvious but for people less tech savy this helps.

I haven't found out whats causing me not to be able to click though. When I searched keybinds in uosc I found that the left click was bound to some uosc specific cmd, could that be the cause?

**Reply by hrxn** [reply]
Could you be more specific?

A single left click does nothing by default in mpv, and as far as I know nothing changes that in uosc.

Tip:
There's a quick and easy way to test your keybinds, simply start mpv like this:
```
mpv --input-test --force-window --idle
```
